src/components/Api.js

```diff
@@ -0,0 +1,136 @@
+export default class Api {
+  constructor({ baseUrl }) {
+    this._baseUrl = baseUrl;
+    this._authToken = "20e48b0c-8946-48f3-99d9-01b588193102";
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: {
+        authorization: this._authToken,
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Great, this was easy to implement, but want to make sure I understand:
When you say pass a reference, I think you mean passing .then(this._checkResponse)
When you say do not pass the call, I think you are warning not to write it this way: .then(this._checkResponse())
Why would the latter not work, since checkResponse is attached to "this", and does not execute until the promise is returned anyway?  I want to understand the difference/significance between reference and call

Reply:
> hy would the latter not work, since checkResponse is attached to "this", and does not execute until the promise is returned anyway? I want to understand the difference/significance between reference and call

`.then(this._checkResponse())` won't work because you need to pass `res` into the call, but you don't get it here

Reply:
Perfect, got it, thank you